packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -272,245 +338,299 @@ export const GridRootStyles = styled('div', {
       justifyContent: 'center',
       color: borderColor,
     },
+    [`& .${c.columnHeaders}`]: {
+      width: 'var(--DataGrid-rowWidth)',
+    },
     '@media (hover: hover)': {
-      [`& .${gridClasses.columnHeaders}:hover`]: columnHeadersStyles,
-      [`& .${gridClasses.columnHeader}:hover`]: columnHeaderStyles,
-      [`& .${gridClasses.columnHeader}:not(.${gridClasses['columnHeader--sorted']}):hover .${gridClasses.sortIcon}`]:
-        {
-          opacity: 0.5,
-        },
+      [`& .${c.columnHeaders}:hover`]: columnHeadersStyles,
+      [`& .${c.columnHeader}:hover`]: columnHeaderStyles,
+      [`& .${c.columnHeader}:not(.${c['columnHeader--sorted']}):hover .${c.sortIcon}`]: {
+        opacity: 0.5,
+      },
     },
     '@media (hover: none)': {
-      [`& .${gridClasses.columnHeaders}`]: columnHeadersStyles,
-      [`& .${gridClasses.columnHeader}`]: columnHeaderStyles,
+      [`& .${c.columnHeaders}`]: columnHeadersStyles,
+      [`& .${c.columnHeader}`]: columnHeaderStyles,
     },
-    [`& .${gridClasses['columnSeparator--sideLeft']}`]: {
+    [`& .${c['columnSeparator--sideLeft']}`]: {
       left: -12,
     },
-    [`& .${gridClasses['columnSeparator--sideRight']}`]: {
+    [`& .${c['columnSeparator--sideRight']}`]: {
       right: -12,
     },
-    [`& .${gridClasses['columnSeparator--resizable']}`]: {
+    [`& .${c['columnSeparator--resizable']}`]: {
       cursor: 'col-resize',
       touchAction: 'none',
       '&:hover': {
-        color: (theme.vars || theme).palette.text.primary,
+        color: (t.vars || t).palette.text.primary,
         // Reset on touch devices, it doesn't add specificity
         '@media (hover: none)': {
           color: borderColor,
         },
       },
-      [`&.${gridClasses['columnSeparator--resizing']}`]: {
-        color: (theme.vars || theme).palette.text.primary,
+      [`&.${c['columnSeparator--resizing']}`]: {
+        color: (t.vars || t).palette.text.primary,
       },
       '& svg': {
         pointerEvents: 'none',
       },
     },
-    [`& .${gridClasses.iconSeparator}`]: {
+    [`& .${c.iconSeparator}`]: {
       color: 'inherit',
     },
-    [`& .${gridClasses.menuIcon}`]: {
-      width: 0,
```

Comment:
The removal of `width: 0` here causes diffs like this:


https://github.com/mui/mui-x/assets/13808724/8fdcab6a-34ec-47e4-b0a1-572844174824

Reply:
It was causing a layout issue that I can't reproduce, so I've restored the `width: 0`.